### ISSUE TITLE
Directives Ignored in Patch Upgrade from v2.3.5 to v2.3.6

### ISSUE THREAD

**Issue by mizoR** [rmosolgo/graphql-ruby#4999]
**Describe the bug**

Hello,

I upgraded graphql-ruby from version v2.3.5 to v2.3.6. I noticed that there are some changes that seem to be incompatible with the previous version, which I didn't expect from a patch version upgrade. 

Specifically, I encountered the following issue:

I have fields with both `directives` and `resolver`/`mutation` settings. In the new version, the content of `directives` is ignored. I resolved the issue by moving the `directives` settings from the field to the resolver, but this change was unexpected.

**Versions**

`graphql` version: v2.3.6
`rails` (or other framework): v7.1.3.4

**Steps to reproduce**

The directive set in the following code will work in v2.3.5, but not in v2.3.6.

```rb
module Types
  class QueryType < Types::BaseObject
    field :sessions_entrance_ceremony,
          resolver: Resolvers::UsersResolver,
          directives: { Directives::Auth => { permission: "users:read" } }
  end
end

module Resolvers
  class UsersResolver < GraphQL::Schema::Resolver
    def resolve
      ...
    end
  end
end
```

In v2.3.6 it works by moving `directives` to `Resolvers::UsersResolver`.

```diff
module Types
  class QueryType < Types::BaseObject
    field :sessions_entrance_ceremony,
          resolver: Resolvers::UsersResolver,
-         directives: { Directives::Auth => { permission: "users:read" } }
  end
end

module Resolvers
  class UsersResolver < GraphQL::Schema::Resolver
+   directive Directives::Auth, permission: "users:read"

    def resolve
      ...
    end
  end
end
```

**Expected behavior**

If this project follows semantic versioning, it may be better to maintain compatibility.

**Comment by rmosolgo** [comment]
Sorry, this was not intentional! I should have made it merge  those instead of clobbering the field directives with the (empty) resolver directives.

**Comment by rmosolgo** [comment]
Thanks for reporting this -- I worked up a patch in #5001 and just released it in v2.3.7. Could you please give that new version a try and let me know if it gives you any more trouble?

**Comment by mizoR** [comment]
Thank you for the quick response and the patch!

I have upgraded to v2.3.7 and tested it in my project. I can confirm that the issue with directives is now resolved and everything works as expected.

I appreciate your prompt assistance in fixing this issue.
